staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -99,13 +100,17 @@ func (f *structuredMergeManager) Update(liveObj, newObj runtime.Object, managed
 	newObjTyped, err := f.typeConverter.ObjectToTyped(newObjVersioned)
 	if err != nil {
 		// Return newObj and just by-pass fields update. This really shouldn't happen.
-		klog.Errorf("[SHOULD NOT HAPPEN] failed to create typed new object of type %v: %v", newObjVersioned.GetObjectKind().GroupVersionKind(), err)
+		atMostEverySecond.Do(func() {
+			klog.Errorf("[SHOULD NOT HAPPEN] failed to create typed new object of type %v: %v", newObjVersioned.GetObjectKind().GroupVersionKind(), err)
+		})
 		return newObj, managed, nil
 	}
 	liveObjTyped, err := f.typeConverter.ObjectToTyped(liveObjVersioned)
 	if err != nil {
 		// Return newObj and just by-pass fields update. This really shouldn't happen.
-		klog.Errorf("[SHOULD NOT HAPPEN] failed to create typed live object of type %v: %v", liveObjVersioned.GetObjectKind().GroupVersionKind(), err)
+		atMostEverySecond.Do(func() {
+			klog.Errorf("[SHOULD NOT HAPPEN] failed to create typed live object of type %v: %v", liveObjVersioned.GetObjectKind().GroupVersionKind(), err)
```

Comment:
I have a little concern about this: `atMostEverySecond` called twice:
If first `atMostEverySecond` is satisfied with ` time.Since(s.lastCall) >= s.delay` so it exec klog message and update the `s.lastCall` which will never let the second `atMostEverySecond` satisfy `time.Since(s.lastCall) >= s.delay`. I don't know if this logic is intended to do.

Reply:
That's exactly what's intended? We don't want the content of the function (the `klog.Errorf`) to be executed more than once per second (and just by-pass/skip it otherwise).

Reply:
If the first atMostEverySecond.Do was called, then the function would have returned anyway, since the next line is "return newObj, managed, nil", so there's no way both of these atMostEverySecond.Do calls would happen for a single request.